karabo_data/geometry2/__init__.py

```diff
@@ -949,23 +967,34 @@ def plot_data(self, modules_data, axis_units='px', frontview=True):
             raise ValueError("axis_units must be 'px' or 'm', not {!r}"
                              .format(axis_units))
 
-        fig = plt.figure(figsize=(10, 10))
-        ax = fig.add_subplot(1, 1, 1)
-        my_viridis = copy(viridis)
-        # Use a dark grey for missing data
-        my_viridis.set_bad('0.25', 1.0)
-
         res, centre = self.position_modules(modules_data)
         min_y, min_x = -centre
         max_y, max_x = np.array(res.shape) - centre
 
-        extent = np.array((min_x - 0.5, max_x + 0.5, min_y - 0.5, max_y + 0.5))
+        _extent = np.array((min_x - 0.5, max_x + 0.5, min_y - 0.5, max_y + 0.5))
         cross_size = 20
         if axis_units == 'm':
-            extent *= self.geom.pixel_size
+            _extent *= self.geom.pixel_size
             cross_size *= self.geom.pixel_size
 
-        ax.imshow(res, origin='lower', cmap=my_viridis, extent=extent)
+        # Use a dark grey for missing data
+        _cmap = copy(viridis)
+        _cmap.set_bad('0.25', 1.0)
+
+        kwargs.setdefault('cmap', _cmap)
+        kwargs.setdefault('extent', _extent)
+        kwargs.setdefault('origin', 'lower')
+
+        if ax is None:
+            fig = plt.figure(figsize=figsize or (10, 10))
+            ax = fig.add_subplot(1, 1, 1)
+
+        im = ax.imshow(res, **kwargs)
+        if cbar:
```

Comment:
This means `cbar={}` is like `cbar=False`. I think if it's a dict containing keyword arguments, I'd expect `{}` to use the defaults, i.e. like True. I don't know if that's what everyone would expect, though.

Reply:
Probably not. I was also thinking it would be nice if people would have access to the colorbar object, in case it need manipulation. I guess that would involve returning the colorbar object with the axis. But I doubt that is is a good solution. Any thoughts ?

Reply:
Yes it does. You can get the colorbar by taking the detour through retrieving the associated image: ```im = ax.images[0]; colorbar = im.colorbar```. I've outlined that in the notebook.